src/sentry_attachment.c

```diff
@@ -156,7 +156,7 @@ sentry__attachments_remove(
     sentry_attachment_t **next_ptr = attachments_ptr;
 
     for (sentry_attachment_t *it = *attachments_ptr; it; it = it->next) {
-        if (attachment_eq(it, attachment)) {
+        if (it == attachment) {
```

Comment:
Yeah, I just wanted to add that for removal, it is no longer helpful to fall back on path comparisons, since there can only be one handle with the same filename inside the list, even for file-based attachments.